### PR TITLE
[FLINK-20885][canal][json] Deserialization exception when using 'canal-json.table.include' to filter out the binlog of the specified table

### PR DESCRIPTION
**Pull request by SteNicholas**
## What is the purpose of the change

*Currently deserialization of `CanalJsonDeserializationSchema` with the option 'canal-json.table.include' to filter out the binlog of the specified table could cause the exception because of the different table schema. Deserialization of `CanalJsonDeserializationSchema` should firstly deserialize to `JsonNode` to extract the 'table' field.*

## Brief change log

  - *Deserialization of `CanalJsonDeserializationSchema` firstly deserializes the message to `JsonNode` to extract the 'table' field for filter.*

## Verifying this change

  - *Add the test binlog record to `canal-data-filter-table.txt` to verfiy the test case `testFilteringTables` in `CanalJsonSerDeSchemaTest`.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by SteNicholas** [comment]
@wuchong , please help to review this bugfix for the deserialization of Canal JSON format.

File: flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowDataDeserializationSchema.java [hunk 1]
```diff
@@ -113,6 +113,36 @@ public RowData deserialize(@Nullable byte[] message) throws IOException {
         }
     }
 
+    public JsonNode deserializeToJsonNode(@Nullable byte[] message) throws IOException {
```
Comment (wuchong) [inline flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowDataDeserializationSchema.java:116]: Could you update `deserialize(..)` to reuse these two method?

**Review by wuchong — COMMENTED** [review]
Thanks @SteNicholas , the PR looks good to me. 

I only have one minor comment.

**Comment by SteNicholas** [comment]
@wuchong , thanks for your detailed review. I have updated `deserialize(...)` to reuse the following two methods. Please help to review again.

**Review by wuchong — COMMENTED** [review]
LGTM.

**Review by wuchong — APPROVED** [review]

**Comment by wuchong** [comment]
There are failed tests.

**Comment by SteNicholas** [comment]
@wuchong , sorry for my mistake to change the behavior message of throwing exception. I have updated the `deserialize(...)` method.

**Pull request merged**